### PR TITLE
Feat/todo tab

### PR DESCRIPTION
**Pull request by EdotJ**
Everything you need is in my commit messages, you can even see the pain I got from making this useless feature and spending too much time on it. Rape me @Kodnot @t-eliks

### CODE REVIEW

**Comment by EdotJ** [comment]
[Add filtering/ordering for TODOs](https://trello.com/c/EXNleNuk/25-add-filtering-ordering-for-todos)

**Comment by EdotJ** [comment]
[Create a separate tab for TODOs](https://trello.com/c/OnScebl2/24-create-a-separate-tab-for-todos)

**Comment by Kodnot** [comment]
Not sure how I feel about reordering todos in the todo tab. As it is now, the user seemingly can reorder only some of the tags, this leads to a confucing experience. Maybe either:
- Disable reordering in the todo tab? Would that be simple?
- Add a faint indication of note borders? Don't like the idea, as in the future if we manage to implement the ordering by date there should be no concept of todos grouped by notes.
- Deal with this in some other way?

File: ToNote/Controls/NotePanel.cs [hunk 1]
```diff
@@ -86,26 +86,43 @@ public string Status
...
+
+               int i = 0;
 
                foreach (var todo in newNoteValue.Todos.OrderBy(x => x.Index))
                {
-                   var todoControl = new TodoControl(todo);
+                    var todoControl = new TodoControl(todo);
 
-                   todoControl.ReadFromFile(todo.FileName);
+                    todoControl.ReadFromFile(todo.FileName);
```
Comment (Kodnot) [inline ToNote/Controls/NotePanel.cs]: [Nitpick] VS supports autoformatting, consider using that.

File: ToNote/Controls/TodoControl.cs [hunk 2]
```diff
@@ -50,7 +51,40 @@ public TodoControl(Todo todo)
             this.AllowDrop = true;
         }
 
-        public Todo Todo { get; private set; }
+        public TodoControl() {
+            this.Loaded += (s, e) =>
```
Comment (Kodnot) [inline ToNote/Controls/TodoControl.cs]: How about extract this into a function instead of copy-pasting it

File: ToNote/Controls/TodoControl.cs [hunk 3]
```diff
@@ -102,5 +136,58 @@ public void TrackKeyword(string keyword, Action action)
             else
                 extendedRTB?.TrackKeyword(keyword, action);
         }
+
+        public static readonly DependencyProperty ListTodosProperty = DependencyProperty.Register("ListTodo",
+           typeof(Todo), typeof(TodoControl), new FrameworkPropertyMetadata(null)
+           { PropertyChangedCallback = (s, e) =>
```
Comment (Kodnot) [inline ToNote/Controls/TodoControl.cs]: The degradation of formatting across these changes is a gruesome representation of a man losing all hope and descending into madness

File: ToNote/Controls/TodoControl.cs [hunk 4]
```diff
@@ -102,5 +136,58 @@ public void TrackKeyword(string keyword, Action action)
...
+
+                    //Looks for an ExtendedRichTextBox with an x:Name="rtb" attribute from the logical children of the generated ContentPresenter which has the DataTemplate applied to it.
+                    var rtb = (ExtendedRichTextBox)template.FindName("rtb", (FrameworkElement)VisualTreeHelper.GetChild(control, 0));
+                    control.ReadFromFile(((Todo)control.DataContext).FileName);
+                    if (rtb == null) return;
+
+                    control.extendedRTB = rtb;
+
+                    //If an ExtendedRichTextBox is found, hooks to the BackspacePressedWhileEmpty event.
+                    rtb.BackspacePressedWithAltShiftModifiers += (o, a) =>
```
Comment (Kodnot) [inline ToNote/Controls/TodoControl.cs]: Again, extract common logic

File: ToNote/Converters/BoolToVisibilityConverter.cs [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+namespace ToNote.Converters
+{
+    public class BoolToVisibilityConverter : IValueConverter
+    {
+        public object Convert(object value, Type targetType, object parameter, System.Globalization.CultureInfo culture)
+        {
+            if (value is bool visibility)
+            {
+                if (visibility)
```
Comment (Kodnot) [inline ToNote/Converters/BoolToVisibilityConverter.cs]: Consider ternary for conciseness

File: ToNote/Views/MainView.xaml.cs [hunk 6]
```diff
@@ -1,7 +1,12 @@
 ﻿namespace ToNote
 {
     using System;
+    using System.Collections;
```
Comment (Kodnot) [inline ToNote/Views/MainView.xaml.cs]: Would be nice if you cleaned up the unused usings. Would be even nicer if you used VS autoformatters and removed unused using solution-wise, as well as formatted the code (ctrl+K, ctrl+D for document, ctrl+K,ctrl+F for selected section by default)

**Review by Kodnot — CHANGES_REQUESTED** [review]

File: ToNote/Controls/NotePanel.cs [hunk 7]
```diff
@@ -79,32 +79,50 @@ public string Status
...
-               {
-                   var rtb = new ExtendedRichTextBox();
+                           rtb.ReadFromFile(file);
 
-                   rtb.ReadFromFile(file);
+                           panel.Items.Add(rtb);
+                       }
+                   }
+                   else
+                       panel.dt.Interval = new TimeSpan(0, 0, 0, 0, 500);
```
Comment (Kodnot) [inline ToNote/Controls/NotePanel.cs]: Consider using `TimeSpan.FromMilliseconds(500)` instead

File: ToNote/Controls/TodoControl.cs [hunk 8]
```diff
@@ -47,10 +47,13 @@ public TodoControl(Todo todo)
                 };
             };
 
-            this.AllowDrop = true;
+            if (IsDragEnabled)
```
Comment (Kodnot) [inline ToNote/Controls/TodoControl.cs]: Dude.

File: ToNote/Controls/TodoControl.cs [hunk 9]
```diff
@@ -102,5 +105,48 @@ public void TrackKeyword(string keyword, Action action)
...
+        public void Load(TodoControl control)
+        {
+            control.Loaded += (s, e) =>
+            {
+                //Gets the DataTemplate defined for the Todo objects.
+                var template = control.ContentTemplate;
+
+                if (template == null) return;
+
+                //Looks for an ExtendedRichTextBox with an x:Name="rtb" attribute from the logical children of the generated ContentPresenter which has the DataTemplate applied to it.
```
Comment (Kodnot) [inline ToNote/Controls/TodoControl.cs:117]: You might need to add `                if (VisualTreeHelper.GetChildrenCount(this) == 0)
                    return;` here

**Review by Kodnot — CHANGES_REQUESTED** [review]

**Comment by Kodnot** [comment]
Also fix the scrollviewer, it should not be visible if the number of elements is small

**Review by Kodnot — APPROVED** [review]

**Pull request merged**